pkg/runtime/framework/core/framework.go

```diff
@@ -112,6 +116,15 @@ func (f *Framework) RunCustomValidationPlugins(oldObj, newObj *trainer.TrainJob)
 	return aggregatedWarnings, aggregatedErrors
 }
 
+func (f *Framework) RunPodNetworkPlugins(info *runtime.Info, trainJob *trainer.TrainJob) error {
```

Comment:
Do we need separate plugin for the Pod Network ?
I can't imagine a use-case when this might be different.

Reply:
I am curious if it is only applied for MPI, should we couple it with MPI plugin for now ?
https://github.com/kubeflow/trainer/blob/8282e748c78b700d0682f91778daeb1bed66f6d2/pkg/runtime/framework/plugins/mpi/mpi.go#L291-L293

In the future, if we see a need for other use-cases when we want to generate endpoints for every Pod, we can refactor it.

Reply:
> I am curious if it is only applied for MPI, should we couple it with MPI plugin for now ?

The endpoint pattern and construction mechanism deeply depend on JobSet functionality (subdomain, etc.) and the JobSet replication parameters. So, MPI plugin can not construct the endpoint list.

Reply:
You are right, so we might need to update the Pipeline Framework to introduce this plugin: https://github.com/kubeflow/trainer/tree/master/docs/proposals/2170-kubeflow-training-v2#pipeline-framework.

I am curious, if in the future we are going to see more decencies between components (e.g. MPI <-> JobSet), do we need to introduce more plugins into Pipeline Framework ?

I guess, the main goal is to have the correct Info object before we run the `ComponentBuilder()` plugin.

Reply:
> You are right, so we might need to update the Pipeline Framework to introduce this plugin: https://github.com/kubeflow/trainer/tree/master/docs/proposals/2170-kubeflow-training-v2#pipeline-framework.

This will be done as a part of https://github.com/kubeflow/trainer/issues/2497

> I am curious, if in the future we are going to see more decencies between components (e.g. MPI <-> JobSet), do we need to introduce more plugins into Pipeline Framework ?

In that case, we can prepare more comprehensive interface like `JobInfrastructure` something, and then add multiple functions like `Network`, `Volume` and so on. However, for now, a single Network interface should be enough.